src/app/clients/getExpandedClientDetails.ts

```diff
@@ -0,0 +1,191 @@
+    packing_date: string;
+    packing_time: string;
+    delivery_instructions: Schema["clients"]["delivery_instructions"];
+    address: string;
+    household: string;
+    "Age & Gender of Children": string;
+    dietary_requirements: string;
+    feminine_products: string;
+    baby_products: Schema["clients"]["baby_food"];
+    pet_food: string;
+    other_requirements: string;
+    extra_information: Schema["clients"]["extra_information"];
+}
+
+export const rawDataToExpandedClientDetails = (
+    rawClientDetails: RawClientDetails
+): ExpandedClientDetails => {
+    if (rawClientDetails === null) {
+        return {
+            "voucher_#": "",
+            full_name: "",
+            phone_number: "",
+            packing_date: "",
+            packing_time: "",
+            delivery_instructions: "",
+            address: "",
+            household: "",
+            "Age & Gender of Children": "",
+            dietary_requirements: "",
+            feminine_products: "",
+            baby_products: null,
+            pet_food: "",
+            other_requirements: "",
+            extra_information: "",
+        };
+    }
+
+    const client = rawClientDetails.client!;
+
+    return {
+        "voucher_#": rawClientDetails.voucher_number,
+        full_name: client.full_name,
+        phone_number: client.phone_number,
+        packing_date: formatDatetimeAsDate(rawClientDetails.packing_datetime),
+        packing_time: formatDatetimeAsTime(rawClientDetails.packing_datetime),
+        delivery_instructions: client.delivery_instructions,
+        address: formatAddressFromClientDetails(client),
+        household: formatHouseholdFromFamilyDetails(client.family),
+        "Age & Gender of Children": formatBreakdownOfChildrenFromFamilyDetails(client.family),
+        dietary_requirements: client.dietary_requirements.join(", "),
+        feminine_products: client.feminine_products.join(", "),
+        baby_products: client.baby_food,
+        pet_food: client.pet_food.join(", "),
+        other_requirements: client.other_items.join(", "),
+        extra_information: client.extra_information,
+    };
+};
+
+export const formatDatetimeAsTime = (datetime: string | null): string => {
+    if (datetime === null || isNaN(Date.parse(datetime))) {
+        return "-";
+    }
+
+    return new Date(datetime).toLocaleTimeString("en-GB");
+};
+
+export const formatAddressFromClientDetails = (
+    client: Pick<
+        Schema["clients"],
+        "address_1" | "address_2" | "address_town" | "address_county" | "address_postcode"
+    >
+): string => {
+    return [
+        client.address_1,
+        client.address_2,
+        client.address_town,
+        client.address_county,
+        client.address_postcode,
+    ]
+        .filter((field) => field)
+        .join(", ");
+};
+
+export const formatHouseholdFromFamilyDetails = (
+    family: Pick<Schema["families"], "age" | "gender">[]
+): string => {
+    let noAdults = 0;
```

Comment:
Can we change this and the one below to numAdults? noAdults currently sounds like a boolean whether or not there is an adult.

Reply:
I've changed the variable names to `adultCount` and `childrenCount`